rules/same-core-dependency-version/same-core-dependency-version.js

```diff
@@ -29,6 +41,14 @@ module.exports = {
     const coreDependencies = getCoreDependencies(process.env.GITHUB_TOKEN);
 
     for (const [dependency, version] of Object.entries(packageJson.dependencies)) {
+      const excludeOption = context.options[0];
+      if (
+        excludeOption && excludeOption.exclude.length > 0 &&
+        excludeOption.exclude.includes(dependency)
+      ) {
+          continue;
+      }
+
       if (!!coreDependencies[dependency] && coreDependencies[dependency] !== version) {
```

Comment:
I suggest joining it with the other if statement:
```ts
const options= context.options[0];     
const ignoredDeps = options && options.exclude || [];
for (const [dependency, version] of Object.entries(packageJson.dependencies)) {
  if (!ignoredDeps.includes(dependency) && !!coreDependencies[dependency] && coreDependencies[dependency] !== version) {
```

Reply:
Nope I would like to keep it separated it is just easier to look 
as this if is about the option "exclude" only